### PR TITLE
zip cudnn with cuda version and cleanup

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge/core, this is ready

**Comment by jakirkham** [comment]
Would it make sense to address this compiler constraint ( https://github.com/conda-forge/nvcc-feedstock/issues/51 ) at the same time?

**Comment by leofang** [comment]
> Would it make sense to address this compiler constraint ( https://github.com/conda-forge/nvcc-feedstock/issues/51 ) at the same time?

That would be great. It's been a blocker for CuPy.

**Comment by leofang** [comment]
Related: #1000

**Comment by isuruf** [comment]
> Would it make sense to address this compiler constraint ( conda-forge/nvcc-feedstock#51 ) at the same time?

No, that cannot be handled with pinning like that, because adding that would add a matrix of compilers for non CUDA builds.

**Comment by isuruf** [comment]
On the other hand, if a feedstock is rerendered with `cuda110` migrator, then the compilers will be handled correctly because we restrict to gcc 7 for all cuda versions.

This is blocking the cuda 11.0 migration, so merge?

**Comment by leofang** [comment]
> On the other hand, if a feedstock is rerendered with `cuda110` migrator, then the compilers will be handled correctly because we restrict to gcc 7 for all cuda versions.

Just curious, where is this pinning done?

**Comment by isuruf** [comment]
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/migrations/cuda110.yaml#L20-L25

**Comment by leofang** [comment]
Thanks @isuruf! LGTM.

File: recipe/migrations/cuda110.yaml [hunk 1]
```diff
@@ -24,26 +24,27 @@ cxx_compiler_version:   # [linux]
...
-#   - cos6   # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
-#   - cos6   # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
-#   - cos6   # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
-#   - cos7   # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+cudnn:
+  - undefined
+  - 7.6                 # [linux64]
+  - 7.6                 # [linux64]
+  - 7.6                 # [linux64]
+  - 7.6                 # [linux64]
```
Comment (kkraus14) [inline recipe/migrations/cuda110.yaml:32]: Should these be 7.6.5 similar to the `conda_build_config.yaml` pinnings?

File: recipe/migrations/windows_cuda.yaml [hunk 2]
```diff
@@ -10,37 +10,20 @@ __migrator:
...
+  - None                       # [win]
+  - 10.0                       # [win]
+  - 10.1                       # [win]
+  - 10.2                       # [win]
+  - 11.0                       # [win]
+cudnn:                         # [win]
+  - undefined                  # [win]
+  - 7.6                        # [win]
+  - 7.6                        # [win]
+  - 7.6                        # [win]
```
Comment (kkraus14) [inline recipe/migrations/windows_cuda.yaml:25]: similarly 7.6.5?

File: recipe/migrations/cuda110.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/migrations/cuda110.yaml]: No, it should be `7` as that's the exports that we use.

File: recipe/migrations/windows_cuda.yaml (hunk 2 above)
Comment (isuruf) [reply recipe/migrations/windows_cuda.yaml]: It should be 7. I've fixed it.

**Review by kkraus14 — APPROVED** [review]

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -44,6 +44,12 @@ cuda_compiler_version:
   - 10.0                       # [linux64]
   - 10.1                       # [linux64]
   - 10.2                       # [linux64]
+cudnn:
+  - undefined
```
Comment (beckermr) [inline recipe/conda_build_config.yaml:48]: I think this needs to be `None`?

Comment (isuruf) [reply recipe/conda_build_config.yaml:48]: It doesn't matter what this value is

Comment (beckermr) [reply recipe/conda_build_config.yaml:48]: Doesn't matter to conda sure, it may matter to the bot however. I need to dig more.

Comment (isuruf) [reply recipe/conda_build_config.yaml:48]: I can change it to None if that helps

Comment (beckermr) [reply recipe/conda_build_config.yaml:48]: Thank you! I think I put in a more general fix that appears to be working.

Comment (beckermr) [reply recipe/conda_build_config.yaml:48]: So no need.

**Pull request merged**